rclcpp/include/rclcpp/clock.hpp

```diff
@@ -78,6 +79,30 @@ class Clock
   Time
   now();
 
+  /**
+   * Sleep until a specified Time, according to clock type.
+   *
+   * Notes for RCL_ROS_TIME clock type:
+   *   - Can sleep forever if ros time is active and received clock never reaches `until`
+   *   - If ROS time enabled state changes during the sleep, this method will immediately return
+   *     false. There is not a consistent choice of sleeping time when the time source changes,
+   *     so this is up to the caller to call again if needed.
+   *
+   * \throws std::runtime_error if the context is invalid
+   * \throws std::runtime_error if `until` has a different clock type from this clock
```

Comment:
Nit, but I think it is more common in our doxygen comments to have `throws` after `return`.

Reply:
Put `\throws` at the end of the doc block in 27b36681